test/common/tls/test_data/certs.sh

```diff
@@ -16,6 +16,73 @@ cleanup() {
     rm ./intermediate_crl_*
 }
 
+
+# $@=<trust domain, CA cert, sequence number> (repeatable for multiple domains and certs per domain)
+generate_spiffe_trust_bundle_mapping() {
+    local trust_domains=()
+
+    # Collecting all trust domain arguments
+    while [[ $# -gt 0 ]]; do
+        trust_domains+=( "$1" "$2" "$3" )
+        shift 3
+    done
+
+    # Use a single redirect for the entire block
+    {
+        echo "{"
+        echo "  \"trust_domains\": {"
+
+        local first_domain=true
+        for (( i=0; i<${#trust_domains[@]}; i+=3 )); do
+            local trust_domain="${trust_domains[i]}"
+            # Quote the array expansion to prevent word splitting
+            local ca_cert_files=( "${trust_domains[i+1]//,/ }" )
+            local sequence_number="${trust_domains[i+2]}"
+
+            if [ "$first_domain" = false ]; then
+                echo "    },"
+            fi
+            first_domain=false
+
+            echo "    \"$trust_domain\": {"
+            echo "      \"sequence_number\": $sequence_number,"
+            echo "      \"keys\": ["
+
+            local first_key=true
+            for ca_cert_file in "${ca_cert_files[@]}"; do
+                # Declare and assign separately
+                local base64_der
+                local modulus
+                base64_der=$(openssl x509 -in "$ca_cert_file" -outform DER | base64 | tr -d '\n\r')
+                modulus=$(openssl x509 -in "$ca_cert_file" -noout -modulus | cut -d'=' -f2 | xxd -r -p | base64 | tr -d "=\n" | tr '/+' '_-')
+
+                if [ "$first_key" = false ]; then
+                    echo ","
+                fi
+                first_key=false
+
+                cat <<EOF
+            {
+              "kty": "RSA",
+              "use": "x509-svid",
+              "x5c": [
+                "$base64_der"
+              ],
+              "n": "$modulus",
+              "e": "AQAB"
+            }
+EOF
+            done
+
+            echo "      ]"
+        done
+
+        echo "    }"
+        echo "  }"
+        echo "}"
+    } > trust_bundles.json
+}
```

Comment:
Could we revert unnecessary test data updates and only keep this function and generated content under the `test/extensions/transport_sockets/tls/cert_validator/spiffe` directly?